### PR TITLE
Minor improvements

### LINKED ISSUE #5: Use temp. variables?
After Adam Wathan: https://youtu.be/crSUWtRYw-M?t=615 (item number `#3`)

```php

                $status = Status::fromStatusLine($line);
                $headers = Headers::fromLines($lines);
                $partials[] = new HttpPartialResponse($status, $headers);

                // 👇 

                $partials[] = new HttpPartialResponse(Status::fromStatusLine($line), Headers::fromLines($lines));
```

What do you think?

### LINKED ISSUE #6: How about moving functions.php?
`functions.php` has the same namespace, so it can live in `/src`.

How about a PR?

### LINKED ISSUE #8: Put errors in conditionals
and do not indent normal execution.

```php
            if (strpos($line, 'HTTP') === 0) {
                $status = Status::fromStatusLine($line);
                $headers = Headers::fromLines($lines);
                $partials[] = new HttpPartialResponse($status, $headers);
            }

			// 👇 

            if (strpos($line, 'HTTP') !== 0) {
				continue;
            }

            $status = Status::fromStatusLine($line);
            $headers = Headers::fromLines($lines);
            $partials[] = new HttpPartialResponse($status, $headers);
```

How about that?

### LINKED ISSUE #9: Specify an Exception Marker Interface for all exception thrown by this library
````php
<?php

interface ClientError implements Throwable
{
}

class SocketErorr extends Exception implements ClientError
{
}
````

So that any consuming application can easily isolate error and exception thrown by the library by typeHinting for `ClientError`

### PR DESCRIPTION
**Pull request by mnavarrocarter**
Added marker exception interface.
Some code cleaning.

Closes #9, Closes #8, Closes #6, Closes #5

### CODE REVIEW

File: src/Redirected.php [hunk 1]
```diff
@@ -19,7 +19,7 @@ interface Redirected
      *
      * The body of those responses is not available.
      *
-     * @return list<PartialResponse>
+     * @return array<PartialResponse>
```
Comment (andrew-demb) [inline src/Redirected.php]: Why?
`list` it is psalm type for numeric array (where keys started from 0 and incrementally increases)

**Comment by szepeviktor** [comment]
You must know there is PHPDoc :)

```
 * This file is part of the https://github.com/mnavarrocarter/php-fetch project. -> @package tag
 * (c) Matías Navarro-Carter <mnavarrocarter@gmail.com> -> @author @copyright tag
 * For the full copyright and license information, please view the LICENSE -> @license tag
 * file that was distributed with this source code.
```

https://docs.phpdoc.org/3.0/guide/references/phpdoc/tags/index.html#tag-reference

**Comment by szepeviktor** [comment]
What do you think @mnavarrocarter?

**Comment by mnavarrocarter** [comment]
Haha, I like it. I'll change it but for today I'll go to rest. I had a busy day at work.

**Pull request merged**